Common/Orders/Slippage/MarketImpactSlippageModel.cs

```diff
@@ -0,0 +1,221 @@
+        /// <param name="eta">coefficient of the temporary impact functio</param>
+        /// <param name="delta">the liquidity scaling factor for permanent impact</param>
+        public MarketImpactSlippageModel(IAlgorithm algorithm, double alpha = 0.891d, double beta = 0.600d,
+                                         double gamma = 0.314d, double eta = 0.142d, double delta = 0.267d)
+        {
+            _algorithm = algorithm;
+            _alpha = alpha;
+            _beta = beta;
+            _gamma = gamma;
+            _eta = eta;
+            _delta = delta;
+        }
+
+        /// <summary>
+        /// Slippage Model. Return a decimal cash slippage approximation on the order.
+        /// </summary>
+        public decimal GetSlippageApproximation(Security asset, Order order)
+        {
+            if (!_symbolDataPerSymbol.TryGetValue(asset.Symbol, out var symbolData))
+            {
+                _symbolDataPerSymbol.Add(asset.Symbol, new SymbolData(_algorithm, asset.Symbol));
+                symbolData = _symbolDataPerSymbol[asset.Symbol];
+            }
+
+            // time taken for execution, we add 700ms to mimic time slippage of filling (convert to by trading day)
+            var time = ((TimeSpan)(_algorithm.UtcTime - order.CreatedTime + TimeSpan.FromMilliseconds(700))).TotalDays * 24d / 6.5d;
+            // expected valid time for impact (+ half an hour)
+            var timePost = time + 1d / 13d;
+            // normalized volume of execution
+            var nu = (double)order.AbsoluteQuantity / time / symbolData.AvgVolume;
+            // liquidity adjustment for temporary market impact, if any
+            var liquidityAdjustment = asset.Fundamentals != null ?
+                                      Math.Pow(asset.Fundamentals.CompanyProfile.SharesOutstanding / symbolData.AvgVolume, _delta) :
+                                      1d;
+            // noise adjustment factor
+            var noise = symbolData.Sigma * Math.Sqrt(timePost);
+
+            // temporary market impact
+            var permImpact = symbolData.Sigma * time * G(nu) * liquidityAdjustment + SampleGaussian(_random) * noise;
+            // permanent market impact
+            var tempImpact = symbolData.Sigma * H(nu) + SampleGaussian(_random) * noise + permImpact * 0.5d;
+            // realized market impact
+            var realizedImpact = tempImpact + permImpact;
+
+            // estimate the slippage by temporary impact
+            return SlippageFromImpactEstimation(realizedImpact);
+        }
+
+        /// <summary>
+        /// The permanent market impact function
+        /// </summary>
+        /// <param name="absOrderQuantity">The absolute, normalized order quantity</param>
+        /// <return>Unadjusted permanent market impact factor</return>
+        protected double G(double absOrderQuantity)
+        {
+            return _gamma * Math.Pow((double)absOrderQuantity, _alpha);
```

Comment:
We can remove the double cast since the `absOrderQuantity` is already a double. Same for `H` method. Both `G` and `H` should be private.